app/api/permissions/routes.ts

```diff
@@ -0,0 +1,55 @@
+import { Application } from 'express';
+import { needsAuthorization } from 'api/auth';
+import { validation } from 'api/utils';
+import { entitiesPermissions } from 'api/permissions/entitiesPermissions';
+import { permissionSchema } from 'shared/types/permissionSchema';
+import { collaborators } from 'api/permissions/collaborators';
+
+export const permissionRoutes = (app: Application) => {
+  app.post(
+    '/api/entities/permissions',
+    needsAuthorization(['admin', 'editor']),
+    validation.validateRequest({
+      properties: {
+        body: {
+          required: ['ids', 'permissions'],
+          properties: {
+            ids: { type: 'array', items: { type: 'string' } },
+            permissions: {
+              type: 'array',
+              items: permissionSchema,
+            },
+          },
+        },
+      },
+    }),
+    async (req, res, _next) => {
+      await entitiesPermissions.setEntitiesPermissions(req.body);
+      res.json(req.body);
+    }
```

Comment:
next should actually be used on the routes to capture possible errors and let the error middleware handle them, same for all the routes on the file.

```
try {
    await entitiesPermissions.setEntitiesPermissions(req.body);
    res.json(req.body);
} catch(e) {
    next(e)
}

Reply:
Thanks, it's done